include/envoy/network/listen_socket.h

```diff
@@ -156,6 +156,17 @@ class ConnectionSocket : public virtual Socket {
    */
   virtual absl::string_view detectedTransportProtocol() const PURE;
 
+  /**
+   * Set requested application protocol(s) (e.g. ALPN in TLS).
+   */
+  virtual void
+  setRequestedApplicationProtocols(const std::vector<absl::string_view>& protocol) PURE;
+
+  /**
+   * @return requested application protocol(s) (e.g. ALPN in TLS), if any.
+   */
+  virtual const std::vector<std::string>& requestedApplicationProtocols() const PURE;
```

Comment:
Note: this returns a list of `std::string` and not of `absl::string_view` like other methods here, because we need to store copy of the data for the lifetime of the connection, and I didn't see the need for a list of `absl::string_view` in addition to the "storage" list of `std::string`, but maybe it makes sense?

Is there a good reason to use `const std::vector<absl::string_view>&` instead of `const std::vector<std::string>&`, here and elsewhere in the code?

Reply:
use of absl::string_view in the codebase is still relatively new, so probably no examples.

I think this is fine as-is. It's easy to change later if it becomes advantageous.